### PR TITLE
* events provider implementation

### PR DESCRIPTION
**Pull request by mikluke**

### CODE REVIEW

File: event/events.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
+package event
+
+import (
+	"errors"
+
+	"github.com/scorum/scorum-go/types"
+)
+
+var wrongEventType = errors.New("wrong type")
```
Comment (megaherz) [inline event/events.go]: please start error names with the `err` prefix

File: event/events.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+		types.VoteOpType:                        toVoteEvent,
+		types.CommentOpType:                     toCommentEvent,
+	}
+}
+
+type Event interface {
+	Type() Type
+}
+
+func ToEvent(op types.Operation, blockID string) (Event, error) {
```
Comment (megaherz) [inline event/events.go]: `(event Event, err error)`

File: provider/utils.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package provider
+
+import "errors"
+
+// MaxRetries is the maximum number of retries before bailing.
+var MaxRetries = 10
```
Comment (megaherz) [inline provider/utils.go]: const?

File: event/events.go [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+	PermLink string
+	Weight   int16
+}
+
+func (e FlagEvent) Type() Type {
+	return FlagEventType
+}
+
+func toVoteEvent(op types.Operation, blockID string) (Event, error) {
+	v, ok := op.(*types.VoteOperation)
```
Comment (megaherz) [inline event/events.go]: Consider `panic` and do not return error at all.

File: provider/provider.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+	SyncInterval time.Duration
+
+	BlocksHistoryMaxLimit  uint32
+	LookupAccountsMaxLimit uint32
+
+	ErrorRetryTimeout time.Duration
+	ErrorRetryLimit   int
+}
+
+var DefaultConfig = Config{
```
Comment (megaherz) [inline provider/provider.go]: 👍

File: provider/provider.go [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+	ErrorRetryTimeout: 10 * time.Second,
+	ErrorRetryLimit:   3,
+}
+
+type Provider struct {
+	Blockchain *scorumgo.Client
+	Config     Config
+}
+
+func NewProvider(url string, config *Config) *Provider {
```
Comment (megaherz) [inline provider/provider.go]: passing nil to use the default config looks a bit strange and unexpected. Please consider an Option patter. See e.g. https://github.com/tmrts/go-patterns/blob/master/idiom/functional-options.md

File: provider/provider.go [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	if config != nil {
+		monitor.Config = *config
+	}
+
+	return monitor
+}
+
+func (p *Provider) Provide(from uint32, types []event.Type, buffer int) (chan event.Event, chan error) {
```
Comment (megaherz) [inline provider/provider.go]: Maybe make the channel read-only ` <- chan event.Event, , <- chan error`

File: event/events.go (hunk 1 above)
Comment (mikluke) [reply event/events.go]: ok

File: provider/utils.go (hunk 3 above)
Comment (mikluke) [reply provider/utils.go]: ok

File: event/events.go [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+		account = v.NewAccountName
+	case *types.AccountCreateByCommitteeOperation:
+		account = v.NewAccountName
+	case *types.AccountCreateWithDelegationOperation:
+		account = v.NewAccountName
+	default:
+		panic(errWrongEventType)
+	}
+
+	return &AccountCreateEvent{
```
Comment (megaherz) [inline event/events.go]: `AccountCreatedEvent`

**Pull request merged**